### ISSUE TITLE
[Bug]: keeps disconnecting 

### ISSUE THREAD

**Issue by vaibhavratnaparkhi** [cjaliaga/home-assistant-aquarea#112]
### The problem

The integration keeps disconnecting and the sensors are disappearing.
![Screenshot_2024-06-09-22-20-19-24_c3a231c25ed346e59462e84656a70e50](https://github.com/cjaliaga/home-assistant-aquarea/assets/149868727/d3bb09c7-6b5c-4a76-bd4e-0a65ea578ad5)

### What version of Home Assistant Core are you running?

core-2024.6.0

### What type of installation are you running?

Home Assistant OS

### What version of the integration are you running?

040700

### Device description

Yes

### Is the integration loading?

Yes

### Is the adapter able to connect to the Aquarea Smart Cloud with a stable internet connection?

Yes

### Are you able to perform the action you're trying to with the Aquarea Smart Cloud Site/App?

Yes

**Comment by lrodorigo** [comment]
Same Here. Reloading the integration does not solve the issue, neither temporarily.

```
  File "/usr/local/lib/python3.12/site-packages/aioaquarea/core.py", line 239, in request
    raise ApiError(error.error_code, error.error_message)
aioaquarea.errors.ApiError: API error: 1000-1503 - Failed communication with adaptor. Please make sure 
it is connected to the network or check again 
after resetting adaptor.
2024-05-11 01:43:38.045 ERROR (MainThread) [custom_components.aquarea.coordinator] Unexpected error fetching aquarea------@gmail.com-B253322680 data: API error: 1000-1503 - Failed communication with adaptor. Please make sure 
it is connected to the network or check again 
after resetting adaptor.
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 318, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/aquarea/coordinator.py", line 67, in _async_update_data

```

**Comment by cjaliaga** [comment]
**TL;DR: Restart the adapter/device.**

That error comes directly from the Aquarea service and it means that the Aquarea cloud is not able to connect to the adapter. I'm glad that **their errors come with the specific information and the instructions to follow.**

The integration works on a device direct mode, which means it asks the Aquarea Service to connect to the adapter and load the latest data **directly from the device** instead of relying on the data cached in the service (which can be outdated, by minutes or even worse). 

In the past I've thought on trying to use the cached data if for whatever reason the integration fails to refresh the data with direct mode, but honestly I've decided to rather show the error as it's better to fix the situation than hide it until data that might be highly outdated.

**Comment by mmmmm009** [comment]
I keep getting the same error:

`Unexpected error fetching aquarea-a.k.xxxxk@xxxxl.com-xxxxx data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 379, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/aquarea/coordinator.py", line 67, in _async_update_data
    await self.device.refresh_data()
  File "/usr/local/lib/python3.13/site-packages/aioaquarea/core.py", line 949, in refresh_data
    self._status = await self._client.get_device_status(self._info.long_id)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/aioaquarea/core.py", line 71, in _wrap
    response = await fn(client, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/aioaquarea/core.py", line 509, in get_device_status
    response = await self.request(
               ^^^^^^^^^^^^^^^^^^^
    ...<4 lines>...
    )
    ^
  File "/usr/local/lib/python3.13/site-packages/aioaquarea/core.py", line 239, in request
    raise ApiError(error.error_code, error.error_message)
aioaquarea.errors.ApiError: API error: 1000-1503 - Failed communication with adaptor. Please make sure 
it is connected to the network or check again 
after resetting adaptor.`


It happens very often:

![szron](https://github.com/user-attachments/assets/ab1cb2b8-7a73-498d-bf92-413d54cc6958)

so, there's nothing that can be done about it?
THX.

**Comment by cjaliaga** [comment]
Hey @mmmmm009 , the recommendation would be to restart the adapter and to ensure the Wi-Fi signal (in case is connected over Wi-Fi instead of Ethernet) is stable and has good coverage where the adapter is located.

**Comment by mmmmm009** [comment]
ok, thx, will try.
So normally it shouldn't disconnect at all?

**Comment by mmmmm009** [comment]
Hi,

I've changed location of heat pump Wifi module. Now it's in line of sight with Wifi router. 
Connection is more stable but still integration disapears quite often.
So maybe the problem is somewhere else?


![Image](https://github.com/user-attachments/assets/7b5b626e-76b7-420e-96cc-fa94107691bb)
